src/parser.ts

```diff
@@ -109,117 +109,112 @@ export const ZodParser = (schema: ZodType<any>) => (
 
   params.seen = params.seen || [];
 
-  const ERROR = new ZodError([]);
-
-  const THROW = () => {
-    RESULT.error = ERROR;
-    throw ERROR;
-  };
+  const ISSUES: ZodIssue[] = [];
 
   const HANDLE = (err: Error) => {
     if (err instanceof ZodError) {
-      ERROR.addIssues(err.issues);
+      ISSUES.push(...err.issues);
       return INVALID;
     }
-    throw ERROR;
+    throw new ZodError(ISSUES);
   };
```

Comment:
This will lose stack trace information. I would recommend reverting your changes to this file, defining error as `let ERROR;`, and then replacing uses of `ERROR` with a `getErrorObject` function that instantiates `ERROR` when it is first called.

Reply:
On second thought, @colinhacks do you figure the stack trace information is useful here? I'm not convinced.